src/lsstseries/analysis/structure_function/bauer_2009a/calculator.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+from lsstseries.analysis.structure_function.base_argument_container import StructureFunctionArgumentContainer
+from lsstseries.analysis.structure_function.base_calculator import StructureFunctionCalculator
+
+
+class Bauer2009AStructureFunctionCalculator(StructureFunctionCalculator):
+    """This class implements the structure function calculation described in
+    Eqn. 4 of Bauer et al. 2009, 2009ApJ...696.1241B [https://arxiv.org/abs/0902.4103]
+
+    `SF(tau) = sqrt( mean(delta_flux^2) - mean(err^2) )`
+
+    Additional references:
+    Graham et al. 2014MNRAS.439..703G [https://arxiv.org/abs/1401.1785]
+    """
+
+    def calculate(self):
+        err2_values = []
+        for lc_idx in range(len(self._time)):
+            lc_times = self._time[lc_idx]
+            lc_fluxes = self._flux[lc_idx]
+            lc_errors = self._err[lc_idx]
+
+            # mask out any nan values
+            t_mask = np.isnan(lc_times)
+            f_mask = np.isnan(lc_fluxes)
+            e_mask = np.isnan(lc_errors)  # always mask out nan errors?
+            lc_mask = np.logical_or(t_mask, f_mask, e_mask)
+
+            lc_times = lc_times[~lc_mask]
+            lc_fluxes = lc_fluxes[~lc_mask]
+            lc_errors = lc_errors[~lc_mask]
+
+            # compute d_times (difference of times) and
+            # d_fluxes (difference of magnitudes, i.e., fluxes) for all gaps
+            # d_times - difference of times
+            dt_matrix = lc_times.reshape((1, lc_times.size)) - lc_times.reshape((lc_times.size, 1))
+            d_times = dt_matrix[dt_matrix > 0].flatten()
+
+            # d_fluxes - difference of fluxes
+            df_matrix = lc_fluxes.reshape((1, lc_fluxes.size)) - lc_fluxes.reshape((lc_fluxes.size, 1))
+            d_fluxes = df_matrix[dt_matrix > 0].flatten()
+
+            # err^2 - errors squared
+            err2_matrix = (
+                lc_errors.reshape((1, lc_errors.size)) ** 2 + lc_errors.reshape((lc_errors.size, 1)) ** 2
```

Comment:
Good question, I asked Neven about this one as well, because the paper being used calls out error as shown in the screen grab below as mean(sigma^2) 

![image](https://user-images.githubusercontent.com/47493171/236921079-5b6a15c4-ac99-4695-9c5f-f1e1fad3ecad.png)

It was not immediately clear to me how to combine the errors from individual measurements, so I asked about that. Neven's response was: "sigma = np.sqrt(sigma_a^2+sigma_b^2)".

So here we add the squares of each measurement error. I don't bother taking the square root of the sum since we square the quantity immediately afterward before finding the mean.

Though, if my algebra is failing me, please raise the flag :)